src/main/java/com/microsoft/sqlserver/jdbc/SQLServerBulkCopy.java

```diff
@@ -2285,6 +2287,12 @@ else if (null != sourceCryptoMeta) {
                             writeNullToTdsWriter(tdsWriter, bulkJdbcType, isStreaming);
                         } else {
                             String colValueStr = colValue.toString();
+                            
+                            // Remove extra trailing zeros added from toString
+                            if (colValue instanceof LocalDateTime || colValue instanceof LocalTime) {
+                            	colValueStr = colValueStr.contains(".") ? colValueStr.replaceAll("0*$","") : colValueStr;
+                            }
```

Comment:
I think you need to run formatter on this

Reply:
Formatted ran. Commited updates.